src/sentry/api/helpers/group_index/update.py

```diff
@@ -738,3 +719,30 @@ def update_groups(
         result["inbox"] = inbox
 
     return Response(result)
+
+
+def handle_is_subscribed(
+    is_subscribed: Any,
+    group_list: Sequence[Group],
+    project_lookup: Dict[int, Any],
+    acting_user: User | None,
+) -> Dict[str, str]:
+    # TODO(dcramer): we could make these more efficient by first
+    # querying for rich rows are present (if N > 2), flipping the flag
+    # on those rows, and then creating the missing rows
+    for group in group_list:
+        # NOTE: Subscribing without an initiating event (assignment,
+        # commenting, etc.) clears out the previous subscription reason
+        # to avoid showing confusing messaging as a result of this
+        # action. It'd be jarring to go directly from "you are not
+        # subscribed" to "you were subscribed due since you were
+        # assigned" just by clicking the "subscribe" button (and you
+        # may no longer be assigned to the issue anyway.)
+        GroupSubscription.objects.create_or_update(
+            user_id=acting_user.id if acting_user else None,
+            group=group,
+            project=project_lookup[group.project_id],
+            values={"is_active": is_subscribed, "reason": GroupSubscriptionReason.unknown},
+        )
```

Comment:
What does it mean to create a subscription if there is no user? Should we even be calling this if that's the case?